### PR TITLE
Initial draft

### PR DESCRIPTION
**Pull request by jotas6**
# Status

- [x] Update README file
- [x] Code examples
- [x] Add content to home page
- [x] Reference sheet
- [x] Instructor notes
- [x] Schedule

### CODE REVIEW

**Comment by jotas6** [comment]
I took inspiration from our tutorials and https://juliadatascience.io/, but I am not sure about what is the best way to give credits here. The same happens for the [Julia workshop](https://github.com/PumasAI-Labs/Julia-Workshop).

File: 04-grouping.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Some times we want to group our data and apply operations according to that grouping
+df = CSV.read("demographics.csv", DataFrame) # Load a fresh copy of our dataset
+
+# The groupby function
+groupby(df, :ISMALE) # Group subjects according to sex
```
Comment (storopoli) [inline 04-grouping.jl:5]: This returns a `GroupedDataFrame` (or something similar if I am not mistaken). Should we care to explain this?

File: 04-grouping.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## More complicated example: transform + group
```
Comment (storopoli) [inline 04-grouping.jl:7]:
```suggestion
## More complicated example: @transform + groupby
```

File: 04-grouping.jl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+@rtransform! df :WEIGHT_cat = :WEIGHT > 70 ? "Over 70 kg" : "Under 70 kg"
+groupby(df, :WEIGHT_cat)
+
+## Tip: groupby can take multiple columns
```
Comment (storopoli) [inline 04-grouping.jl:11]:
```suggestion
## Tip: groupby can take multiple columns as grouping keys
```

File: 04-grouping.jl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+groupby(df, [:ISMALE, :WEIGHT_cat]) # Now we get 4 groups
+
+# Combining (@combine)
```
Comment (storopoli) [inline 04-grouping.jl:14]: Maybe Summarizing instead of Combining?
What you thin?

**Review by storopoli — CHANGES_REQUESTED** [review]
Some changes requests and also some suggestions.

File: 03-transform.jl [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# Apply some transformation to one or more columns in our data
+include("02-select.jl")
+
+# Change the sex encoding (ISMALE)
+df
+@transform df :SEX = [i == 0 ? "Female" : "Male" for i in :ISMALE] # Create a new column
+@transform df :ISMALE = [i == 0 ? "Female" : "Male" for i in :ISMALE] # Modify an existing column
```
[the remark this answers is not in the document]
Comment (jotas6) [reply 03-transform.jl:7]: I am not sure if this is enough or what we are looking for, but this part might be related: https://github.com/PumasAI-Labs/Julia-Workshop/blob/cca736d6de5ff0d815834c9ec8defb61b7d16b76/01-variables/04-containers.jl#L3C1-L6C29

[the remark this answers is not in the document]
Comment (storopoli) [reply 03-transform.jl:7]: Yeah, then we need to either change this part or explain the `[f(i) for i in collection]` syntax.

[the remark this answers is not in the document]
Comment (jotas6) [reply 03-transform.jl:7]: I included a short explanation of array comprehensions in the loops section, but I don't know if it is detailed enough: https://github.com/PumasAI-Labs/Julia-Workshop/blob/cca736d6de5ff0d815834c9ec8defb61b7d16b76/02-syntax/02-loops.jl#L50C1-L57C46

I could add more examples there or modify this example if that would be better.

File: 04-grouping.jl (hunk 4 above)
Comment (jotas6) [reply 04-grouping.jl:14]: I think that is a better name. I went with the same name as the macro, but summarizing is probably more descriptive

File: 04-grouping.jl (hunk 1 above)
Comment (jotas6) [reply 04-grouping.jl:5]: Yes, it returns a `GroupedDataFrame` and it looks like it is possible to manipulate the results (https://dataframes.juliadata.org/stable/lib/functions/#DataFrames.groupby), but I was afraid that might be too much information. 

I also think that the main use case for `groupby` is to split data and then use `combine` or `@combine`, but I could definitely work on some examples that involve using `GroupedDataFrame`s.

**Comment by jotas6** [comment]
Sorry about forgetting to add `.jl` at the end of the package names

File: 04-grouping.jl (hunk 1 above)
Comment (storopoli) [reply 04-grouping.jl:5]: Yes, then don't mention it. Maybe in the instructor's notes?

**Comment by storopoli** [comment]
> Sorry about forgetting to add `.jl` at the end of the package names

Yeah specially around `DataFrames.jl` ecosystem is very confusing.

File: 04-grouping.jl (hunk 1 above)
Comment (jotas6) [reply 04-grouping.jl:5]: Yes, I agree. I will add a note about it in the instructor notes

**Pull request merged**